File: scrape_score.py
import sys
import json
import time

try:
    import socketio
except ImportError:
    print("Error: The 'python-socketio' package is not installed.")
    print("Please install it using: pip install \"python-socketio[client]<5.0.0\"")
    sys.exit(1)

# Check version because the server is Socket.IO v2
# We need python-socketio version 4.x
sio_version = getattr(socketio, "__version__", "")
if sio_version.startswith("5."):
    print(f"Warning: You are using python-socketio v{sio_version}.")
    print("The target server uses Socket.IO v2, which may not be compatible with v5+ clients.")
    print("If connection fails, please downgrade: pip install \"python-socketio[client]<5.0.0\"")

sio = socketio.Client(logger=False, engineio_logger=False)

ROOM_ID = 35317902
URL_ENDPOINT = "https://sportsscore24.com"

@sio.event
def connect():
    print("[SUCCESS] Connected to live score server!")
    # Send subscription message just like the website does
    sio.emit('subscribe', {
        'type': 1,
        'rooms': [ROOM_ID]
    })
    print(f"[*] Subscribed to room {ROOM_ID}. Waiting for live data...\n")

@sio.event
def update(msg):
    # Depending on the server/client, data might come as a JSON string or dict
    if isinstance(msg, str):
        try:
            data = json.loads(msg)
        except json.JSONDecodeError:
            print("[ERROR] Failed to parse message:", msg)
            return
    else:
        data = msg
        
    if isinstance(data, dict) and data.get("type") == 1 and "data" in data:
        score_data = data["data"]
        
        # Extract properties
        team1 = score_data.get("spnnation1", "Team 1")
        score1 = score_data.get("score1", "Yet to bat")
        team2 = score_data.get("spnnation2", "Team 2")
        score2 = score_data.get("score2", "Yet to bat")
        status = score_data.get("spnballrunningstatus") or score_data.get("spnmessage")
        balls = score_data.get("balls", [])
        
        print("="*40)
        print("         LIVE MATCH UPDATE")
        print("="*40)
        print(f"{team1: <20} : {score1}")
        print(f"{team2: <20} : {score2}")
        print("-"*40)
        
        if status:
            print(f"Status: {status}")
            
        if balls:
            print(f"Recent Balls: {', '.join(map(str, balls))}")
            
        print("="*40)
        print("\n")

@sio.event
def connect_error(err):
    print("[ERROR] Connection error:", err)

@sio.event
def disconnect():
    print("[!] Disconnected from server.")

def main():
    print(f"[*] Attempting to connect to {URL_ENDPOINT} for room {ROOM_ID}...")
    try:
        # The site specifies polling and websocket transports with standard socket.io path
        sio.connect(
            URL_ENDPOINT, 
            socketio_path="/socket.io/", 
            transports=['websocket', 'polling']
        )
        sio.wait()
    except Exception as e:
        print("[ERROR] Could not connect:", e)
        print("Make sure you are using python-socketio version 4.x.")

if __name__ == '__main__':
    main()

File: scrape_score.js
const io = require("socket.io-client");

// The room ID taken from the live score URL (?room=35317902)
const ROOM_ID = 35317902;
// The Websocket Server URL found in the source code
const url = "https://sportsscore24.com";

console.log(`[*] Connecting to live score server at ${url}...`);

// Important: The server uses Socket.IO v2, so you MUST use socket.io-client v2.x
// Install with: npm install socket.io-client@2
const socket = io(url, {
    path: "/socket.io/",
    transports: ["websocket", "polling"],
    reconnection: true,
    reconnectionAttempts: 10,
    reconnectionDelay: 3000,
    timeout: 20000,
    forceNew: true,
});

socket.on("connect", () => {
    console.log("[SUCCESS] Connected to server!");

    // The server requires this exact 'subscribe' emit format to join the room
    socket.emit("subscribe", {
        type: 1,
        rooms: [ROOM_ID],
    });

    console.log(`[*] Subscribed to room ${ROOM_ID}. Waiting for real-time updates...\n`);
});

socket.on("update", (msg) => {
    try {
        // Message can be a JSON string or an object directly
        const data = typeof msg === "string" ? JSON.parse(msg) : msg;

        // We only care about type 1 messages which contain the scorecard
        if (data.type === 1 && data.data) {
            const scoreData = data.data;

            console.log("========================================");
            console.log("           LIVE MATCH UPDATE            ");
            console.log("========================================");

            const team1 = scoreData.spnnation1 || "Team 1";
            const team2 = scoreData.spnnation2 || "Team 2";
            const score1 = scoreData.score1 || "Yet to bat";
            const score2 = scoreData.score2 || "Yet to bat";

            console.log(`${team1.padEnd(20)} : ${score1}`);
            console.log(`${team2.padEnd(20)} : ${score2}`);
            console.log("----------------------------------------");

            const status = scoreData.spnballrunningstatus || scoreData.spnmessage;
            if (status) {
                console.log(`Status: ${status}`);
            }

            if (scoreData.balls && scoreData.balls.length > 0) {
                console.log(`Recent Balls: ${scoreData.balls.join(', ')}`);
            }

            console.log("========================================\n");
        }
    } catch (err) {
        console.error("[ERROR] Parse error:", err.message);
    }
});

socket.on("connect_error", (err) => {
    console.error("[ERROR] Connection error:", err.message);
});

socket.on("disconnect", (reason) => {
    console.log("[!] Disconnected from server:", reason);
});
